cmd/verify.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright © 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/hacbs-contract/ec-cli/cmd/internal/applicationsnapshot"
+	rs "github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"github.com/spf13/cobra"
+)
+
+var snapshot rs.ApplicationSnapshotSpec
+
+// verifyCmd represents the verify command
+var verifyCmd = &cobra.Command{
+	Use:   "verify",
+	Short: "Change me",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		var jsonstr string
+		// Fail if input and filepath are used.
+		if arguments.input != "" && arguments.filepath != "" {
+			fmt.Println("Use either --input or --filepath")
+			os.Exit(1)
+		}
+
+		// Filepath handler, reads the file and stores the content as string
+		if arguments.filepath != "" {
+			file, err := ioutil.ReadFile(arguments.filepath)
+
+			if err != nil {
+				fmt.Println(err)
+				os.Exit(1)
+			}
+
+			jsonstr = string(file)
+		}
+
+		// Input handler
+		if arguments.input != "" {
+			jsonstr = arguments.input
+		}
+
+		// Validate the json, exit on failure
+		if err := applicationsnapshot.Validate(jsonstr); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Unmarshall json into struct, exit on failure
+		if err := json.Unmarshal([]byte(jsonstr), &snapshot); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		report, err := applicationsnapshot.GenerateReport(snapshot)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		fmt.Println(report)
+		os.Exit(0)
+	},
```

Comment:
In eval we do `RunE` instead and return the err, this is not nice and we have [HACBS-633](https://issues.redhat.com/browse/HACBS-633) to clean that up...